ambry-server/src/integration-test/java/com.github.ambry.server/ServerTestUtil.java

```diff
@@ -712,6 +743,48 @@ protected static void endToEndReplicationWithMultiNodeMultiPartitionTest(int int
       }
     }
 
+    // ttl update all blobs and wait for replication
+    List<Thread> threads = new ArrayList<>(blobIds.size());
+    AtomicReference<Throwable> exception = new AtomicReference<>(null);
+    Map<BlockingChannel, List<BlobId>> channelToBlobIds = new HashMap<>();
+    for (int i = 0; i < blobIds.size(); i++) {
+      final BlobId blobId = blobIds.get(i);
+      if (i % 3 == 0) {
+        channelToBlobIds.computeIfAbsent(channel1, updateChannel -> new ArrayList<>()).add(blobId);
+      } else if (i % 3 == 1) {
+        channelToBlobIds.computeIfAbsent(channel2, updateChannel -> new ArrayList<>()).add(blobId);
+      } else {
+        channelToBlobIds.computeIfAbsent(channel3, updateChannel -> new ArrayList<>()).add(blobId);
+      }
+    }
+    for (Map.Entry<BlockingChannel, List<BlobId>> entry : channelToBlobIds.entrySet()) {
```

Comment:
could also consider replacing lines 760-778 with:
```java
    channelToBlobIds.entrySet().stream().map(entry -> CompletableFuture.supplyAsync(() -> {
      try {
        for (BlobId blobId : entry.getValue()) {
          updateBlobTtl(entry.getKey(), blobId);
        }
        return null;
      } catch (Throwable e) {
        throw new RuntimeException("Exception updating ttl for: " + entry, e);
      }
    })).forEach(CompletableFuture::join);
```

CompletableFuture::supplyAsync runs the tasks in a common thread pool and CompletableFuture::join will throw an exception if the task threw an exception

Reply:
Done !